Incorrect output in tests: `test_reduce1d` and `test_reduce_layouts`

Example:
```bash
E           AssertionError: 
E           Not equal to tolerance rtol=0.01, atol=0.001
E           
E           Mismatched elements: 1 / 1 (100%)
E           Max absolute difference: 41.38921
E           Max relative difference: 0.84127945
E            x: array(90.58714, dtype=float32)
E            y: array([[49.197933]], dtype=float32)
```

Other errors can be viewed in the CI logs: https://github.com/intel/intel-xpu-backend-for-triton/actions/runs/12982420553/job/36202215178

--- a/lib/Conversion/TritonGPUToLLVM/Utility.cpp
+++ b/lib/Conversion/TritonGPUToLLVM/Utility.cpp
@@ -15,7 +15,7 @@
 static int __builtin_clz(unsigned x) {
   unsigned long r;
   _BitScanReverse(&r, x);
-  return static_cast<int>(r);
+  return static_cast<int>(r ^ 31);
 }
 
 static int __builtin_ctz(unsigned x) {

